packages/lesswrong/components/ea-forum/EAGBanner.tsx

```diff
@@ -0,0 +1,201 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { Components, registerComponent } from "../../lib/vulcan-lib";
+import { useCurrentUser } from "../common/withUser";
+import { AnalyticsContext, useTracking } from "../../lib/analyticsEvents";
+import { useCookiesWithConsent } from "../hooks/useCookiesWithConsent";
+import moment from "moment";
+import ForumNoSSR from "../common/ForumNoSSR";
+import { HIDE_EAG_BANNER_COOKIE } from "@/lib/cookies/cookies";
+import { Link } from "@/lib/reactRouterWrapper";
+import { useUserLocation } from "@/lib/collections/users/helpers";
+import { distance } from "../community/modules/LocalGroups";
+import { getCachedUserCountryCode } from "../common/CookieBanner/geolocation";
+import { lightbulbIcon } from "../icons/lightbulbIcon";
+
+const styles = (theme: ThemeType) => ({
+  root: {
+    display: "flex",
+    justifyContent: "space-between",
+    columnGap: "10px",
+    padding: '10px 10px 10px 12px',
+    backgroundColor: theme.palette.grey[200],
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    fontSize: 15,
+    lineHeight: '22px',
+    fontWeight: 450,
+    borderRadius: theme.borderRadius.default,
+    marginBottom: 20,
+  },
+  lightbulb: {
+    flex: 'none',
+    height: 30,
+    width: 30,
+    alignSelf: 'center',
+    color: theme.palette.buttons.alwaysPrimary,
+  },
+  content: {
+    flexGrow: 1,
+  },
+  topRow: {
+    display: 'flex',
+    alignItems: 'center',
+    columnGap: 6,
+    fontSize: 13,
+    lineHeight: '17px',
+    opacity: 0.5,
+    marginBottom: 2,
+  },
+  infoIcon: {
+    fontSize: 14,
+    transform: 'translateY(2px)',
+  },
+  bold: {
+    fontWeight: 700,
+  },
+  link: {
+    textDecoration: 'underline',
+  },
+  applyLink: {
+    color: theme.palette.buttons.alwaysPrimary,
+    fontWeight: 600,
+    textDecoration: 'underline',
+    textUnderlineOffset: '3px',
+    whiteSpace: 'nowrap',
+    '&:hover': {
+      textDecoration: 'underline',
+      opacity: 0.75,
+    },
+  },
+  close: {
+    flex: 'none',
+    fontSize: 20,
+    cursor: "pointer",
+    marginLeft: 13,
+    "&:hover": {
+      opacity: 0.75,
+    },
+  },
+});
+
+// This is the data for the next EAGx (Toronto, CA)
+const eagName = 'EAGxToronto'
+const eagLocation = {
+  lat: 43.6532,
+  lng: -79.3832,
+}
+const eagCountry = 'CA'
+const eagPostLink = "/events/WGeby2GfMHH8jXmMY/eagxtoronto"
+const eagLink = "https://www.effectivealtruism.org/ea-global/events/eagxtoronto-2024"
+const applicationDeadline = moment('2024-07-31', 'YYYY-MM-DD')
```

Comment:
After this test, if we want to iterate on this banner, we can move these somewhere where the Events Team can edit it. We'll probably want to make things like the distance radius editable as well.